stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -744,6 +770,76 @@ end
     return matches
 end
 
+# Identify an argument being completed in a method call. If the argument is empty, method
+# suggestions will be provided instead of argument completions.
+function identify_possible_method_completion(partial, last_idx)
+    fail = 0:-1, Expr(:nothing), 0:-1, 0
+
+    # First, check that the last punctuation is either ',', ';' or '('
+    idx_last_punct = something(findprev(x -> ispunct(x) && x != '_' && x != '!', partial, last_idx), 0)::Int
+    idx_last_punct == 0 && return fail
+    last_punct = partial[idx_last_punct]
+    last_punct == ',' || last_punct == ';' || last_punct == '(' || return fail
+
+    # Then, check that `last_punct` is only followed by an identifier or nothing
+    before_last_word_start = something(findprev(in(non_identifier_chars), partial, last_idx), 0)
+    before_last_word_start == 0 && return fail
+    all(isspace, @view partial[nextind(partial, idx_last_punct):before_last_word_start]) || return fail
+
+    # Check that `last_punct` is either the last '(' or placed after a previous '('
+    frange, method_name_end = find_start_brace(@view partial[1:idx_last_punct])
+    method_name_end ∈ frange || return fail
+
+    # Strip the preceding ! operators, if any, and close the expression with a ')'
+    s = replace(partial[frange], r"\G\!+([^=\(]+)" => s"\1"; count=1) * ')'
+    ex = Meta.parse(s, raise=false, depwarn=false)
+    isa(ex, Expr) || return fail
+
+    # `wordrange` is the position of the last argument to complete
+    wordrange = nextind(partial, before_last_word_start):last_idx
+    return frange, ex, wordrange, method_name_end
+end
+
+# Provide completion for keyword arguments in function calls
+function complete_keyword_argument(partial, last_idx, context_module)
+    frange, ex, wordrange, = identify_possible_method_completion(partial, last_idx)
+    fail = Completion[], 0:-1, frange
+    ex.head === :call || is_broadcasting_expr(ex) || return fail
+
+    kwargs_flag, funct, args_ex, kwargs_ex = _complete_methods(ex, context_module, true)::Tuple{Int, Any, Vector{Any}, Set{Symbol}}
+    kwargs_flag == 2 && return fail # one of the previous kwargs is invalid
+
+    methods = Completion[]
+    complete_methods!(methods, funct, Any[Vararg{Any}], kwargs_ex, -1, kwargs_flag == 1)
+    # TODO: use args_ex instead of Any[Vararg{Any}] and only provide kwarg completion for
+    # method calls compatible with the current arguments.
```

Comment:
In theory, this is a pretty bad choice of methods to consider, since it is specifically only looking at methods without kwargs. And you never want to pass `-1` here, as you can accidentally break interactivity for several minutes. So without those bad trailing args, and looking only at methods with kwargs, we want something like this:

```julia
# get the list of possible kw method table
kwt = try; Core.kwftype(funct); catch ex; ex isa ErrorException || rethrow(); return fail; end
ml = Completion[]
complete_methods!(ml, kwt, Any[Any, funct, args_ex...])
ml = Method[m.method for m in ml if m isa MethodCompletion]
# decide if we might have a `...` in the list
have_kwarg = any(isa(TextCompletion), ml)
for m in ml
    kwli = m.method
    slotnames = ccall(:jl_uncompress_argnames, Vector{Symbol}, (Any,), kwli.slot_syms)
    have_kwarg = any(endswith("..."), slotnames)
    have_kwarg && break
end
if have_kwarg
    # if we have a `kwarg...`, instead return the union of all possible method kwargs for this function
    methods = Base.MethodList(kwt.name.mt)
end
# now get the kwarg names, and filter them to just the matching ones
kwargs = Set{String}()
for m in ml
    kwli = m.method
    slotnames = ccall(:jl_uncompress_argnames, Vector{Symbol}, (Any,), kwli.slot_syms)
    union!(kwargs, filter!(slotnames))
end
```

(kwargs_ex should not be passed here either, as that is sort of just an historical accident of past implementation problems)